src/Orleans/Messaging/GatewayManager.cs

```diff
@@ -253,7 +253,8 @@ private void UpdateLiveGatewaysSnapshot(IEnumerable<Uri> currentKnownGateways, T
 
         public void Dispose()
         {
-            gatewayRefreshTimer?.Dispose();
+            if (gatewayRefreshTimer != null)
+                gatewayRefreshTimer.Dispose();
```

Comment:
potential race condition. Opinions @ReubenBond since you added this?
I would prefer to just ignore the warning explicitly. Hopefully they'll fix this at some point (see https://github.com/dotnet/roslyn-analyzers/issues/291)